### ISSUE TITLE
Emulator should not accept hyphen in instance name

### ISSUE THREAD

**Issue by gsouf** [GoogleCloudPlatform/cloud-spanner-emulator#13]
Here is the allowed characters for a spanner instance:

> Instance ID must start with a lowercase letter, be 2-64 characters long, contain only lowercase letters, numbers, or hyphens, and not end with a hyphen

However the spanner emulator will let us create instance with an underscore.

On live spanner:

```
gcloud spanner instances create some_test  \
  --config=emulator-config \
  --description="Test Instance" \
  --nodes=1

ERROR: (gcloud.spanner.instances.create) INVALID_ARGUMENT: Invalid instance ID: `some_test`.  Instance ID must start with a lowercase letter, be 2-64 characters long, contain only lowercase letters, numbers, or hyphens, and not end with a hyphen.
```

Same command on emulator will work and will add the instance

**Comment by Vizerai** [comment]
Thanks for pointing this out. We are working on a fix that should be incorporated in the next release.

**Comment by jacobdufault-google** [comment]
Fixed by b0c06e0e17cac56af7d3335c18dfb1a780ff89d4
